control-plane/agents/src/bin/core/controller/wrapper.rs

```diff
@@ -972,13 +975,29 @@ impl ClientOps for Arc<tokio::sync::RwLock<NodeWrapper>> {
             Err(SvcError::GrpcRequestError { source, .. })
                 if source.code() == tonic::Code::NotFound =>
             {
-                Ok(())
+                Err(SvcError::PoolNotFound {
+                    pool_id: request.id.clone(),
+                })
             }
             Err(error) => Err(error),
             Ok(_) => Ok(()),
         }
     }
 
+    /// Import a pool on the node via gRPC.
+    async fn import_pool(&self, request: &ImportPool) -> Result<PoolState, SvcError> {
+        let dataplane = self.grpc_client_locked(request.id()).await?;
+        let import_response = dataplane.import_pool(request).await;
+        let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
+        self.update_pool_states(ctx.deref_mut()).await?;
+        self.update_replica_states(ctx.deref_mut()).await?;
+        let pool = self.read().await.pool(&request.id);
+        match (pool, import_response) {
+            (_, Ok(pool)) => Ok(pool),
+            (_, Err(error)) => Err(error),
```

Comment:
seems we're never using pool that we got from self?

Reply:
I think I will remove the line 994?

Reply:
Done.